src/os/bluestore/BlueStore.cc

```diff
@@ -12333,22 +12339,24 @@ void BlueStore::_zoned_cleaner_thread() {
   ceph_assert(!zoned_cleaner_started);
   zoned_cleaner_started = true;
   zoned_cleaner_cond.notify_all();
-  std::deque<uint64_t> zones_to_clean;
+  auto a = dynamic_cast<ZonedAllocator*>(shared_alloc.a);
+  auto f = dynamic_cast<ZonedFreelistManager*>(fm);
+  const auto *zones_to_clean = a->zoned_get_zones_to_clean();
   while (true) {
-    if (zoned_cleaner_queue.empty()) {
+    if (a->get_num_zones_to_clean() == 0) {
```

Comment:
IMO more elegant would be to call zoned_get_zones_to_clean() at this point.  Which should check the atomic internally and return either nullptr or a pointer to a container instance.

Reply:
Done.